src/MX/MegaMenu/Model/MenuRepository.php

```diff
@@ -127,4 +157,28 @@ public function deleteById($menuId)
     {
         return $this->delete($this->getById($menuId));
     }
+
+    /**
+     * Truncate tables
+     *
+     * @param MenuInterface $menu
+     * @param string $tableName
+     */
+    public function truncateTables(MenuInterface $menu)
```

Comment:
move the logic from this method to the menu resource model and rename this method to something like deleteAll without any parameter so it will be clear this method deletes all menus not just the one which is passed as argument